src/agent.rs

```diff
@@ -165,3 +179,38 @@ impl Agent {
         output.clone()
     }
 }
+
+fn check_cipher_blacklist(cipher: &str, shim: &str) -> bool {
```

Comment:
Can you open a follow-up so this function doesn't have to read the file every time?

Reply:
I already implemented a solution. The blacklist is now loaded once at the start of a test run and a reference passed on to every agent so it can check ciphers against it.